src/core.js

```diff
@@ -2433,23 +2433,28 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
    * @param {number} column Visual column index.
    * @param {string} key Property name.
    * @param {string} value Property value.
+   * @fires Hooks#beforeSetCellMeta
    * @fires Hooks#afterSetCellMeta
    */
   this.setCellMeta = function(row, column, key, value) {
-    let physicalRow = row;
-    let physicalColumn = column;
+    const allowSetCellMeta = instance.runHooks('beforeSetCellMeta', row, column, key, value);
 
-    if (row < this.countRows()) {
-      physicalRow = this.toPhysicalRow(row);
-    }
+    if (allowSetCellMeta !== false) {
```

Comment:
```suggestion
    if (allowSetCellMeta === false) {
      return;
    }
```
I'd suggest terminating executing the logic inside the method body by placing `return`  here.  IMO it'll increase the maintainability of the code especially when the code between bracket starts to grow.

Additionally, I think that the coordinates such as `row` and `column` should be the same for both hooks. Now `beforeSetCellMeta` gets raw values and `afterSetCellMeta` gets processed by index mapper.

Reply:
I'm not sure. This method creates `physicalRow` and `physicalColumn` based on `row` and `column`. Next, it processed by index mapper for both `physical` variables, but at the end `afterSetCellMeta` gets raw `row` and `column` too.

Reply:
You are right! Both hooks right now get the raw value 👍